lib/srv/db/common/reporting.go

```diff
@@ -0,0 +1,272 @@
+	r.engine.SendError(err)
+}
+
+func (r *reportingEngine) HandleConnection(ctx context.Context, sessionCtx *Session) error {
+	activeConnections.With(r.getLabels()).Inc()
+	defer activeConnections.With(r.getLabels()).Dec()
+
+	start := r.clock.Now()
+	defer func() {
+		connectionDurations.With(r.getLabels()).Observe(r.clock.Since(start).Seconds())
+	}()
+
+	return trace.Wrap(r.engine.HandleConnection(ctx, sessionCtx))
+}
+
+var _ Engine = &reportingEngine{}
+
+var commonLabels = []string{teleport.ComponentLabel, "db_protocol", "db_type"}
+
+func getLabels(component string, db types.Database) prometheus.Labels {
+	return map[string]string{
+		teleport.ComponentLabel: component,
+		"db_protocol":           db.GetProtocol(),
+		"db_type":               db.GetType(),
+	}
+}
+
+func (r *reportingEngine) getLabels() prometheus.Labels {
+	return getLabels(r.component, r.database)
+}
+
+var (
+	clientReadBytes = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_client_read_bytes_total",
+			Help:      "Total number bytes read from the DB clients",
+		},
+		commonLabels,
+	)
+
+	clientWrittenBytes = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_client_written_bytes_total",
+			Help:      "Total number bytes written to the DB clients",
+		},
+		commonLabels,
+	)
+
+	messagesFromClient = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_messages_from_client_total",
+			Help:      "Number of messages (packets) received from the DB client",
+		},
+		commonLabels,
+	)
+
+	messagesFromServer = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_messages_from_server_total",
+			Help:      "Number of messages (packets) received from the DB server",
+		},
+		commonLabels,
+	)
+
+	methodCallCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_method_call_count_total",
+		},
+		append([]string{"method"}, commonLabels...),
+	)
+
+	methodCallLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_method_call_latency",
+			// lowest bucket start of upper bound 0.001 sec (1 ms) with factor 2
+			// highest bucket start of 0.001 sec * 2^15 == 32.768 sec
+			Buckets: prometheus.ExponentialBuckets(0.001, 2, 16),
+		},
+		append([]string{"method"}, commonLabels...),
+	)
+
+	initializedConnections = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_initialized_connections_total",
+			Help:      "Number of initialized DB connections",
+		},
+		commonLabels,
+	)
+
+	activeConnections = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "db_active_connections",
```

Comment:
```suggestion
			Name:      "db_active_connections_total",
```

nit: suffix describing unit ([reference](https://prometheus.io/docs/practices/naming/#metric-names)).